small-talk/app/chat/page.js

```diff
@@ -5,159 +5,128 @@ import socket from "../../util/socket";
-	const onUsernameSelection = (username) => {
-		socket.auth = { username };
-        socket.connect();
-    };
-
-	// // just added a button to check if connecting with backend
-    const handleConnectButtonClick = () => {
-        onUsernameSelection("randomusername");
-    };
+	const handleConnectButtonClick = () => {
+		onUsernameSelection("randomusername");
+	};
 
 	const handleSendMessage = () => {
-        sendMessage(inputMessage, selectedUser, socket, setPrivateMessages, setInputMessage);
-    };
-
-	return (
-        <ThemeLayout>
-            <div // all the styles added in this file are temporary for testing purposes
-				data-testid="chat-container" 
-				style={{ 
-					display: 'flex',
-					flexDirection: 'column',
-					alignItems: 'center', // Align items to the center horizontally
-					backgroundColor: 'pink',
-					color: 'black',
-					padding: '20px',
-					borderRadius: '10px',
-					maxWidth: '80%', // Limit the maximum width of the container
-					margin: '20vh auto 0', // Move the container down by 20% of viewport height
-					height: '70vh', // Set the height to 70% of the viewport height
-					maxHeight: '600px', // Limit the maximum height to 600px
-					overflow: 'hidden', // Hide overflow to prevent scrollbars on the container itself
-					boxShadow: '0 0 10px rgba(0, 0, 0, 0.1)', // Add a subtle shadow for depth
-            	}}>
-
-				<h2>Private Chat</h2>
-				<div 
-					className="private-messages-list"
-					style={{
-						height: 'calc(100% - 220px)', // Set the height of the message container dynamically
-						width: '100%',
-						overflowY: 'auto', // Allow vertical scrolling if content overflows
-					}}>
-
-					{privateMessages.map((message, index) => (
-						<div key={index}>
-						<strong>{message.from === socket.id ? "You" : "Recipient"}:</strong> {message.content}
-						</div>
-					))}
-				</div>
-
-				<h3 style = {{ color:'blue' }}>Users:</h3>
-				<ul className="user-list" style={{ color: 'purple', padding: 0, margin: 0, maxHeight: '120px', overflowY: 'auto' }}>
-					{user.map(user => (
-						<li key={user.userID} onClick={() => setSelectedUser(user)}>
-							{user.username}
-						</li>
-					))}
-				</ul>
+		sendMessage(inputMessage, selectedUser, setPrivateMessages, setInputMessage);
+	};
 
-				<input
-					type="text"
-					value={inputMessage}
-					onChange={(e) => setInputMessage(e.target.value)}
-					placeholder="Type your message here..."
-					style={{ width: '100%', marginBottom: '10px' }} // Set the width to 100% and add some bottom margin
-				/>
-				<div style={{ display: 'flex', justifyContent: 'center' }}>
-					<button style={{ backgroundColor: 'yellow', marginRight: '10px' }} onClick={handleConnectButtonClick}>Connect</button>
-					<button style={{ backgroundColor: 'green', marginRight: '10px' }} onClick={handleSendMessage}>Send</button>
-				</div>
-
-            </div>
+	const onUsernameSelection = (username) => {
```

Comment:
Question about this. With the useEffect. Is it connecting to the socket as 'random' then again here as username?

Reply:
I believe the problem with useEffect is being connected twice. We might need to work on that issue soon after justin's pr.